x/wasm/internal/keeper/keeper.go

```diff
@@ -328,6 +341,16 @@ func (k Keeper) generateContractAddress(ctx sdk.Context, codeID uint64) sdk.AccA
 	return addrFromUint64(contractID)
 }
 
+func (k Keeper) GetLastID(ctx sdk.Context, lastIDKey []byte) uint64 {
```

Comment:
I would remove the second arg and just hardcode it to `types.KeyLastCodeID`, as that is all it is used for.

Also, this should be called `GetNextID`, as there is no contract yet for this ID (I believe)

Reply:
Was trying to keep API similarity to `autoIncrementID`.  I feel `autoIncrementID` should be split into two specialized functions that only take in `ctx`.  Thoughts?

Reply:
Yeah, that sounds good. I saw your fix and I like it